### PR TITLE
Assembly: Add Assembly to precommit

### PR DESCRIPTION
**Pull request by chennes**
After this merge, anyone working on Assembly should make sure they have installed pre-commit into their local clone of the repo.  https://freecad.github.io/DevelopersHandbook/gettingstarted/ (see item 3 in "Setting up for development")

### CODE REVIEW

File: src/Mod/Assembly/InitGui.py [hunk 1]
```diff
@@ -51,8 +51,7 @@ class AssemblyWorkbench(Workbench):
     def __init__(self):
         print("Loading Assembly workbench...")
         self.__class__.Icon = (
-            FreeCAD.getResourceDir()
-            + "Mod/Assembly/Resources/icons/AssemblyWorkbench.svg"
+            FreeCAD.getResourceDir() + "Mod/Assembly/Resources/icons/AssemblyWorkbench.svg"
```
Comment (PaddleStroke) [inline src/Mod/Assembly/InitGui.py:54]: Nice the 100 characters limit seems more forgiving than the standard black that Brad applied.

**Review by PaddleStroke — COMMENTED** [review]
Ok, to further work on assembly we must opt-in this pre-commit right ? Though I see that sketcher is in the list and I have installed nothing regarding this.

File: src/Mod/Assembly/InitGui.py (hunk 1 above)
Comment (chennes) [reply src/Mod/Assembly/InitGui.py:54]: We don't have a system in place to force the pre-commit hook (eventually we'll add a CI step that requires the formatting be correct, but that's not implemented right now). So actually installing pre-commit in the repo is a step that every developer must take.

Comment (PaddleStroke) [reply src/Mod/Assembly/InitGui.py:54]: By the way regarding this pre-commit hook: It makes PR very hard to review as every file you touch get completely re-writen. I'm thinking about FlachyJoe's PR that were +-1500 lines when the real code was less than 100 lines.
So perhaps all files in the sources should be passed through a formating step now. Though that would make horrible conflicts to all the opened PR. Though the pre-commit hook only dilute the problem of making conflicts to other open PR.

**Pull request merged**